src/reconcile/rec_write.c

```diff
@@ -1819,10 +1813,12 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
     if (F_ISSET(r, WT_REC_IN_MEMORY))
         goto copy_image;
 
-    /*
-     * If there are saved updates, do history store eviction.
-     */
-    if (multi->supd != NULL) {
+    /* If we need to restore the page to memory, copy the disk image. */
+    if (r->cache_write_restore && multi->supd != NULL) {
```

Comment:
Doesn't `r->cache_write_restore` imply that `multi->supd != NULL`?

Reply:
Not always when the page splits. It's possible one of the page after split doesn't have any saved updates.